lnprototest/backend/bitcoind.py

```diff
@@ -71,16 +72,29 @@ def __init__(self, basedir: str):
             "-logtimestamps",
             "-nolisten",
         ]
-        self.port = reserve()
         self.btc_version = None
-        logging.debug("Port is {}, dir is {}".format(self.port, self.bitcoin_dir))
+
+    def __reserve(self) -> int:
+        """
+        When python will ask for a free port for the os, it is possible that
+        with concurrence access the port that python picks a port that it is free
```

Comment:
```suggestion
        with concurrent access, the port that is picked is a port that is not free
```

Reply:
maybe I am misunderstanding this @vincenzopalazzo , but from the stack overflow link you have referenced, i think these suggestions make the docstring easier to understand

Reply:
are you suggesting to keep only the link?

Reply:
not really .I think the docstring is still relevant ,although it could be clearer with those suggested changes.